rust/src/wrapper/format.rs

```diff
@@ -176,15 +178,32 @@ pub struct Format {
     pub(crate) inner: Arc<Mutex<xlsx::Format>>,
 }
 
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(typescript_type = "Format")]
+    pub type JsFormat;
+}
+
 #[wasm_bindgen]
 impl Format {
     /// Create a new Format object.
     ///
     /// Create a new Format object to use with worksheet formatting.
+    /// If a Format obejct is provided as an argument, it will be cloned.
     #[wasm_bindgen(constructor)]
-    pub fn new() -> Format {
-        Format {
-            inner: Arc::new(Mutex::new(xlsx::Format::new())),
+    pub fn new(format: Option<JsFormat>) -> WasmResult<Format> {
+        if let Some(format) = format {
+            // HACK: Since wasm-bindgen does not supports Option<&RustStruct>, we cast JsValue as Format.
+            let format = utils::format_of_jsval(&format).ok_or(XlsxError::Type(
+                "Expected a value of type Format".to_string(),
+            ))?;
+            // Clone the inner object
+            let format = format.deep_clone();
+            Ok(format)
+        } else {
+            Ok(Format {
+                inner: Arc::new(Mutex::new(xlsx::Format::new())),
+            })
         }
     }
```

Comment:
I'm now thinking that implementing `deep_clone` as a new method might be a better approach because:

* Neither JavaScript nor Rust supports function overloading.
  * We might want to add parameters to the constructor in the future.
* The code should be clearer and more straightforward with new method